Oop №2   composition

LarionovKirill opened the pull request.

LarionovKirill committed:
Добавление лабы№1 по ООП. (Я ее делал в другом проекте до индивидуальной, поэтому одним коммитом. )

LarionovKirill committed:
.

LarionovKirill committed:
Fix(OOP): Исправление недочетов в проекте.

LarionovKirill committed:
Fix(Uml лаба 1.jpg):

LarionovKirill committed:
Fix(Uml лаба 1.jpg): Исправил uml - диаграмму.

LarionovKirill committed:
Начало лабы №2.
Feature(Category.cs): добавлено перечисление.
Refactoring (Item.cs): теперь категориии являются композицией данного класса.
Feature (ItemsTab.cs): добавлена работа с перечислениями в клиентский код

LarionovKirill committed:
Feature(ItemsTabs.cs): Добавил обрабоку выбора категории и их изменение.

LarionovKirill committed:
Merge pull request #13 from LarionovKirill/OOP-№1

Oop №1

LarionovKirill committed:
Я опять не посмотрел в какой ветке работаю. Но я слил с нужной веткой и убрал все изменения. Благо их было немного.

LarionovKirill committed:
Fix(ItemsTab.cs): исправлен маленький баг с удалением.

LarionovKirill committed:
Feature(Address.cs): создал класс адреса пользователя.
Feature(Valuevalidator.cs): Добавил метод для проверки индекса.

LarionovKirill committed:
Fix(Project): везде уменьшил обращение к пространству имен.

LarionovKirill committed:
Fix(project): Исправил пространства имен.

LarionovKirill committed:
.

LarionovKirill committed:
Fix(Customer.cs): исправена передача имени свойства в метод.

LarionovKirill committed:
Fix(Address.cs): исправена передача имени свойства.
Fix(Customer.cs): исправена передача имени свойства.
Fix(Item.cs): исправена передача имени свойства.

LarionovKirill committed:
Feature(AddressControl.cs): добавил контрол для адреса и внедрил его в клиентский код.

LarionovKirill committed:
Feature(CustomerTab.cs): добавил возможность изменять имена.

LarionovKirill committed:
Feature(AddressControl.cs): Временный коммит

LarionovKirill committed:
.

LarionovKirill committed:
Feature(AddressControl.cs): добавил нормально элемент управления с верными значениями адреса.

LarionovKirill committed:
Feature(Store.cs): добавил класс Store.
Feature(CustomerTab.cs): изменил обращение к списку.

LarionovKirill committed:
Feature (uml лаба 2.jpg): добавил диаграмму классов проекта.

LarionovKirill committed:
Fix(ItemTab.Designer.cs): чуть подправил размер.

LarionovKirill committed:
/

LarionovKirill committed:
/

LarionovKirill committed:
/

LarionovKirill committed:
Feature(gitignore)

LarionovKirill committed:
Delete slnx.sqlite

LarionovKirill committed:
Delete VSWorkspaceState.json

LarionovKirill committed:
Delete .suo

LarionovKirill committed:
Merge pull request #20 from LarionovKirill/tempOOP1

Temp oop1

LarionovKirill committed:
Feature(gitignore)

LarionovKirill committed:
Merge pull request #21 from LarionovKirill/OOP-№2---Composition

Feature(gitignore)

LarionovKirill committed:
Delete slnx.sqlite

LarionovKirill committed:
Delete VSWorkspaceState.json

LarionovKirill committed:
Delete .suo

LarionovKirill committed:
Merge pull request #22 from LarionovKirill/tempOOP2

Temp oop2

LarionovKirill committed:
Fix(Tabs): Везде постарался добавить адаптивную верстку(Если что напиши что исправить)
Fix(AddressControl.cs): Вроде решил баг с дизайнером. Убрал глобальные переменные, но если что-то объявиться, то скажи.

Ershovoy commented on ObjectOrientedPractics/OOP/Model/Category.cs:
```diff
@@ -0,0 +1,22 @@
+    /// <summary>
+    /// Перечисление, содержащее категории товаров.
+    /// </summary>
+    public enum Category
```
Лучше уточнить, так как категория слишком общее слово, она может быть не только у товара.
```suggestion
    public enum ItemCategory
```

Ershovoy commented on ObjectOrientedPractics/OOP/Model/Category.cs:
```diff
@@ -0,0 +1,22 @@
+        Sport,
+        Auto,
+        Child,
+        books
```
Потерял заглавную букву у books.

Ershovoy commented on ObjectOrientedPractics/OOP/Model/Customer.cs:
```diff
@@ -0,0 +1,94 @@
+	/// Класс хранит пользователя
+	/// </summary>
+	[Serializable]
+	public class Customer
```
У класса **Customer** и **Item** для табуляции используются табы, а в остальных файлах проекта четыре пробела. Оставь что-нибудь одно.

Ershovoy commented on ObjectOrientedPractics/OOP/View/Tabs/AddressControl.cs:
```diff
@@ -0,0 +1,199 @@
+{
+    public partial class AddressControl : UserControl
+    {
+        /// <summary>
+        /// Копия формы для передачи данных
+        /// </summary>
+        public CustomerTab _customerTab;
+
+        /// <summary>
+        /// Хранит преедан пользователь или нет.
+        /// </summary>
+        private bool _currentCustomer = false;
```
Обращение к закрытым полям класса -- зло, даже если внутри. Я уже говорил. Замени на авто свойство.
```suggestion
        /// <summary>
        /// Копия формы для передачи данных
        /// </summary>
        public CustomerTab CustomerTab { get; set; };

        /// <summary>
        /// Хранит преедан пользователь или нет.
        /// </summary>
        private bool CurrentCustomer { get; set; } = false;
```

LarionovKirill committed:
Fix(Customer.cs):Убрал лишнее поле.
Fix(item.cs):Убрал лишнее поле и изменил тип данных стоимости.
Fix(ValueValidetor.cs):Изменил типы данных параметров метода.
Fix(CustomerTab.cs):Добавил XML, изменил методы с валидацией.
Fix(ItemsTab.cs):Изменил методы с валидацией и убрал лишний метод.

Ershovoy commented on ObjectOrientedPractics/OOP/Diagrams/Uml лаба 2.jpg:
Выровняй классы на UML диаграмме относительно друг друга, чтобы было минимально возможное число пересичений. Также расположи их в логическом порядке относительно друг друга. Слева сверху -- общее. Всё остальное, что относительно него правее и ниже -- его частное. Вот пример удачного расположения
![image](https://github.com/LarionovKirill/Programming/assets/50287099/bea42001-8c97-40b8-acd9-fd8008a31bb0)

LarionovKirill replied:
Ок, исправлю

LarionovKirill committed:
Update ObjectOrientedPractics/OOP/Model/Category.cs

Co-authored-by: Ilya Ershov <50287099+Ershovoy@users.noreply.github.com>

LarionovKirill committed:
Update ObjectOrientedPractics/OOP/View/Tabs/AddressControl.cs

Co-authored-by: Ilya Ershov <50287099+Ershovoy@users.noreply.github.com>

LarionovKirill committed:
Fix(Category.cs): Добавил заглавную букву)

LarionovKirill committed:
Fix(model): Вставил код еще раз. VS должна отворматировать сама на табы. Не знаю как они там стали. Может сам как-то сделал не заметил.

LarionovKirill committed:
Fix(): исправил последствия автосвойств.

LarionovKirill committed:
Merge branch 'OOP-1-temp' into OOP2-temp

LarionovKirill committed:
Fix(addresControl.cs): добавил автосвойство.

LarionovKirill committed:
/

LarionovKirill committed:
/

LarionovKirill committed:
Fix(itemsTab.cs): испраивл название лейбла

LarionovKirill committed:
Merge branch 'OOP2-temp' into OOP-№2---Composition

LarionovKirill committed:
Merge pull request #29 from LarionovKirill/OOP-№2---Composition

Oop №2   composition

LarionovKirill committed:
/

LarionovKirill committed:
Исправлены баги с валидацией

LarionovKirill committed:
Merge pull request #30 from LarionovKirill/OOP2-temp

Oop2 temp

LarionovKirill committed:
Fix(ItemsTab.cs): исправил баг с валидацией.

LarionovKirill committed:
Fix(ItemsTab.cs): исправил верстку.

LarionovKirill committed:
Fix(Diagrams): исправил UML диаграмму
